### PR TITLE
Fix issues in files with Scala 2.13 specific syntax

### PR DESCRIPTION
**Pull request by tgodzik**
Fixes https://github.com/scalameta/metals/issues/1664

Previously, we would use 2.12 parser and tokenizer by default. Now we define the default dialect as 2.13 in one place and make sure it's used everywhere.

### CODE REVIEW

File: tests/slow/src/test/scala/tests/feature/DefinitionCrossLspSuite.scala [hunk 1]
```diff
@@ -19,6 +19,36 @@ class DefinitionCrossLspSuite
...
 
+  test("underscore") {
+    cleanDatabase()
+    for {
+      _ <- server.initialize(
+        s"""
+           |/metals.json
+           |{
+           |  "a": {
+           |    "scalaVersion": "${BuildInfo.scala213}"
```
Comment (kpbochenek) [inline tests/slow/src/test/scala/tests/feature/DefinitionCrossLspSuite.scala:30]: This is unrelated to switching to default scala 2.13, correct?
Or you wanted to test that default scala would be 2.13 by veryfing 2.13 only feature would be parsed?

Comment (tgodzik) [reply tests/slow/src/test/scala/tests/feature/DefinitionCrossLspSuite.scala:30]: That is related, I wanted to make sure we properly parse 2.13 syntax.
